src/formatting/items.rs

```diff
@@ -1642,12 +1643,28 @@ fn rewrite_type<R: Rewrite>(
     if let Some(ty) = rhs {
         // If there's a where clause, add a newline before the assignment. Otherwise just add a
         // space.
-        if !generics.where_clause.predicates.is_empty() {
+        let has_where = !generics.where_clause.predicates.is_empty();
+        if has_where {
             result.push_str(&indent.to_string_with_newline(context.config));
-        } else {
-            result.push(' ');
         }
-        let lhs = format!("{}=", result);
+
+        let lhs = match between {
+            Some(between)
+                if contains_comment(&context.snippet_provider.span_to_snippet(between)?) =>
+            {
+                let comment_shape = Shape::indented(indent, context.config)
+                    .block_left(context.config.tab_spaces())?;
+                combine_strs_with_missing_comments(
+                    context,
+                    &result,
+                    "=",
+                    between,
+                    comment_shape,
+                    true,
+                )?
+            }
+            _ => format!("{}{}=", result, if has_where { "" } else { " " }),
+        };
```

Comment:
I think you'll find this to produce incorrect formatting in cases where the ident is not the final element in the lhs. 

Adding a comment before the assignment operator from the previous assignment will illustrate this:

```rust
type A<'a: 'static, T: Ord + 'static>: Eq + PartialEq where T: 'static + Copy /* x */ = Vec<u8>;
```

This ties into my comment above about not accepting an arbitrary 'between" span as the function param that defers to callers to define the span. Once that's changed to the full item span, the correct between span for this particular instance can calculated (and encapsulated) within this function based on the presence/absence of other items.

It also makes it easier in the future to support not dropping comments between other things (like the `type` keyword and the ident e.g. `type /*abc*/ A = u8;`)